src/calendar/__tests__/Calendar.test.tsx

```diff
@@ -0,0 +1,170 @@
+        <CalendarButton {...state} goto="nextMonth" data-testid="next-month">
+          next month
+        </CalendarButton>
+        <CalendarButton {...state} goto="nextYear">
+          next year
+        </CalendarButton>
+      </div>
+
+      <CalendarGrid {...state} as="table" className="dates">
+        <thead>
+          <tr data-testid="weekDays">
+            {state.weekDays.map((day, dayIndex) => {
+              return (
+                <CalendarWeekTitle
+                  {...state}
+                  as="th"
+                  scope="col"
+                  key={dayIndex}
+                  dayIndex={dayIndex}
+                >
+                  <abbr title={day.title}>{day.abbr}</abbr>
+                </CalendarWeekTitle>
+              );
+            })}
+          </tr>
+        </thead>
+        <tbody>
+          {state.daysInMonth.map((week: any[], weekIndex: React.Key) => (
+            <tr key={weekIndex}>
+              {week.map((day: Date, dayIndex: React.Key) => (
+                <CalendarCell {...state} as="td" key={dayIndex} date={day}>
+                  <CalendarCellButton {...state} date={day} />
+                </CalendarCell>
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </CalendarGrid>
+    </CalendarWrapper>
+  );
+};
+
+describe("Calendar", () => {
+  it("should render correctly", () => {
+    const { getByTestId: testId } = render(
+      <CalendarComp defaultValue={"10-7-2020"} />,
+    );
+
+    expect(testId("weekDays").children).toHaveLength(7);
+    expect(testId("current-year")).toHaveTextContent("October 2020");
+  });
+
+  it("should have proper calendar header keyboard navigation", () => {
+    const { getByTestId: testId, getByText: text } = render(
+      <CalendarComp defaultValue={"10-7-2020"} />,
+    );
+
+    expect(testId("current-year")).toHaveTextContent("October 2020");
+    press.Tab();
+    press.Enter();
+    expect(text("previous year")).toHaveFocus();
+    expect(testId("current-year")).toHaveTextContent("October 2019");
+    press.Tab();
+    press.Enter();
+    expect(text("previous month")).toHaveFocus();
+    expect(testId("current-year")).toHaveTextContent("September 2019");
+    press.Tab();
+    press.Enter();
+    expect(text("next month")).toHaveFocus();
+    expect(testId("current-year")).toHaveTextContent("October 2019");
+    press.Tab();
+    press.Enter();
+    expect(text("next year")).toHaveFocus();
+    expect(testId("current-year")).toHaveTextContent("October 2020");
+  });
+
+  it("should proper grid navigation", () => {
+    const { getByTestId: testId, getByLabelText: label } = render(
+      <CalendarComp defaultValue={"10-7-2020"} />,
+    );
+
+    expect(testId("current-year")).toHaveTextContent("October 2020");
+    press.Tab();
+    press.Tab();
+    press.Tab();
+    press.Tab();
+    press.Tab();
+
+    expect(label("Today, Wednesday, October 7, 2020 selected")).toHaveFocus();
```

Comment:
Test fails today, because todays date changes.
Select a date that's not today.

Reply:
lol my bad, these test are designed to work only once per year. :p 😆